meta-sokol-flex-staging/recipes-kernel/lttng/lttng-tools/0001-Ensure-that-the-consumerd-configure-arguments-are-us.patch

```diff
@@ -1,17 +1,53 @@
-From 0d09ed08a84ec77e01bede24847f4ced4e99057c Mon Sep 17 00:00:00 2001
-From: Christopher Larson <chris_larson@mentor.com>
-Date: Fri, 29 Sep 2023 17:07:30 -0500
+From e371a77f6dae4d797b7b20b8f0058c7893934a48 Mon Sep 17 00:00:00 2001
+From: Arslan Ahmad <arslan_ahmad@mentor.com>
+Date: Thu, 19 Oct 2023 15:46:05 +0500
 Subject: [PATCH] Ensure that the consumerd configure arguments are used
 
-Currently, the --with-consumerd32-bin, --with-consumerd32-libdir, etc arguments
-are not being used by lttng-sessiond, which causes problems with providing a
-well configured multilib tracing environment. Ensure that they are used as the
-default values in sessiond.
+Currently, the --with-consumerd32-bin, --with-consumerd32-libdir, etc
+arguments are not being used by lttng-sessiond, which causes problems
+with providing a well configured multilib tracing environment. Ensure
+that they are used as the default values in sessiond. Also ensure that
+the paths are present before spawning them.
 
 Signed-off-by: Christopher Larson <chris_larson@mentor.com>
 ---
-Upstream-Status: Pending
+ src/bin/lttng-sessiond/client.c          | 7 +++++--
+ src/bin/lttng-sessiond/sessiond-config.c | 8 ++++----
+ src/common/defaults.h                    | 4 ++++
+ 3 files changed, 13 insertions(+), 6 deletions(-)
 
+diff --git a/src/bin/lttng-sessiond/client.c b/src/bin/lttng-sessiond/client.c
+index c798234..763bf8a 100644
+--- a/src/bin/lttng-sessiond/client.c
++++ b/src/bin/lttng-sessiond/client.c
+@@ -1002,6 +1002,7 @@ static int process_client_msg(struct command_ctx *cmd_ctx, int *sock,
+ 		int *sock_error)
+ {
+ 	int ret = LTTNG_OK;
++	struct stat st;
+ 	bool need_tracing_session = true;
+ 	bool need_domain;
+ 	bool need_consumerd;
+@@ -1285,7 +1286,8 @@ static int process_client_msg(struct command_ctx *cmd_ctx, int *sock,
+ 			/* 64-bit */
+ 			pthread_mutex_lock(&the_ustconsumer64_data.pid_mutex);
+ 			if (the_config.consumerd64_bin_path.value &&
+-					the_ustconsumer64_data.pid == 0 &&
++					stat(the_config.consumerd64_bin_path.value, &st) == 0 &&
```

Comment:
if user provided wrong path as argument to sessiond, we'll be ignoring it instead of properly erroring out?

Reply:
Good point! Both kernel consumerd and sessiond paths show errors if it fails to find a valid path or has a path it can't access, whereas the 32 and 64 bit consumerd will end up getting an execl() failure with a less pleasant error message for the user. This change makes it silent rather than trying and failing to spawn, which is worse. I'll drop this, since there's no further error checking on these values done elsewhere, better to let the spawn fail than to show the user no feedback. Ideally it'd probably check for invalid or missing paths earlier on in the config handling.

Reply:
Good point! Both kernel consumerd and sessiond paths show errors if it fails to find a valid path or has a path it can't access, whereas the 32 and 64 bit consumerd will end up getting an execl() failure with a less pleasant error message for the user. This change makes it silent rather than trying and failing to spawn, which is worse. I'll drop this, since there's no further error checking on these values done elsewhere, better to let the spawn fail than to show the user no feedback. Ideally it'd probably check for invalid or missing paths earlier on in the config handling.